src/App/Fossa/LicenseScanner.hs

```diff
@@ -349,7 +347,7 @@ licenseScanSourceUnit vendoredDependencyScanMode baseDir vendoredDeps = do
 
   -- finalizeLicenseScan takes archives without Organization information. This orgID is appended when creating the build on the backend.
   -- We don't care about the response here because if the build has already been queued, we get a 401 response.
-  finalizeLicenseScan $ ArchiveComponents $ NE.toList archives
+  finalizeLicenseScan $ ArchiveComponents{archives = NE.toList archives, forceRebuild = vendoredDependencyScanMode == SkippingDisabledViaFlag}
```

Comment:
We should avoid using the `{a = foo, b = bar}` constructor style, especially when the types of the fields are different, and the type system can enforce the correct usage.
```suggestion
  finalizeLicenseScan $ ArchiveComponents (NE.toList archives) (vendoredDependencyScanMode == SkippingDisabledViaFlag)
```